### PR TITLE
Use Regex to get endpoints to validate

### PR DESCRIPTION
**Pull request by jparoly**
Change-Id: I38a3dfe844428e98968e11f61b69f82059cb950f

### CODE REVIEW

File: tests/kuttl/common/assert-sample-deployment.yaml [hunk 1]
```diff
@@ -205,10 +205,23 @@ status:
...
+      template='{{.status.apiEndpoints.ironic.public}}{{":"}}{{.status.apiEndpoints.ironic.admin}}{{":"}}{{.status.apiEndpoints.ironic.internal}}{{"\n"}}'
+      regex="http:\/\/ironic-public-openstack\.apps.*:http:\/\/ironic-admin-openstack\.apps.*:http:\/\/ironic-internal-openstack\.apps.*"
+      apiEndpoints=$(oc get -n openstack ironics.ironic.openstack.org ironic -o go-template="$template")
+      matches=$(echo "$apiEndpoints" | sed -e "s?$regex??")
+      if [ -z "$matches" ]; then
+        exit 0
+      else
+        exit 1
+      fi
   - script: |
```
Comment (jparoly) [inline tests/kuttl/common/assert-sample-deployment.yaml:225]: I don't think i can have two scripts but not sure.

File: tests/kuttl/common/assert-sample-deployment.yaml [hunk 2]
```diff
@@ -205,10 +205,23 @@ status:
...
       set -x
       RETURN_CODE=0
```
Comment (jparoly) [inline tests/kuttl/common/assert-sample-deployment.yaml:227]: I'm no understanding why this fails in non crc environments.  I don't see any crc specific things.

**Review by steveb — APPROVED** [review]

**Pull request merged**